lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2856,7 +2856,7 @@ mkApiTransaction
     -> Lens' (ApiTransaction n) (Maybe ApiBlockReference)
     -> IO (ApiTransaction n)
 mkApiTransaction
-    ti txid mfee ins cins outs ws (meta, timestamp) txMeta setTimeReference = do
+    ti txid mfee cins ins outs ws (meta, timestamp) txMeta setTimeReference = do
```

Comment:
Ooh, this looks like a very likely place to make a mistake, is there anyway we can distinguish between the two types of [(TxIn, Maybe TxOut)] that makes sense in this context?

Reply:
@sevanspowell I agree that it would be easy for a caller to get these arguments muddled up, especially given that the types are identical.

I've created a record type `MkApiTransactionParams`, and altered `mkApiTransaction` to take this record instead of all the positional parameters. Hopefully this should make it easier at call sites to determine which parameter is which.